packages/amplify_auth_cognito/lib/src/CognitoSignUp/cognito_user_attributes.dart

```diff
@@ -1,24 +1,185 @@
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:flutter/foundation.dart';
+
 /// Standard user attributes available for configuring via `Amplify.Auth.signUp`,
 /// or updating via `Amplify.Auth.updateUserAttribute` and
 /// `Amplify.Auth.updateUserAttributes`.
 ///
 /// More information about these attributes can be found
 /// [here](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html).
-abstract class CognitoUserAttributes {
-  static const address = 'address';
-  static const email = 'email';
-  static const familyName = 'family_name';
-  static const gender = 'gender';
-  static const givenName = 'given_name';
-  static const locale = 'locale';
-  static const middleName = 'middle_name';
-  static const name = 'name';
-  static const nickname = 'nickname';
-  static const phoneNumber = 'phone_number';
-  static const picture = 'picture';
-  static const preferredUsername = 'preferred_username';
-  static const profile = 'profile';
-  static const updatedAt = 'updated_at';
-  static const website = 'website';
-  static const zoneinfo = 'zoneinfo';
+@immutable
+class CognitoUserAttributes extends UserAttributeKey {
```

Comment:
This should be singular, `CognitoUserAttribute`.

Preferably, it should be `CognitoUserAttributeKey`

This definitely isn't a well definied pattern we have established, but in general the Cognito specific classes have the same name as the abstract class they extend, with a prefix of `Cognito`. I think this is a nice pattern to follow. Since the public API docs for methods like `updateUserAttribute()` will show the abstract class `UserAttributeKey`, users have to do a bit of guess work (or reference the docs) to know what the actual class name is. That guess work is a little easier if we consistently prefix the abstract class with `Cognito`.